bootstrap/src/stream.ml

```diff
@@ -0,0 +1,576 @@
+open Rudiments
+
+type 'a elm =
+  | Nil
+  | Cons of 'a * 'a t
+and 'a t = 'a elm lazy_t
+
+let empty = lazy Nil
+
+let init n ~f =
+  let rec fn i n f = lazy begin
+    match i < n with
+    | false -> Nil
+    | true -> begin
+      let elm = f i in
+      let t = fn (Usize.succ i) n f in
+      Cons(elm, t)
+    end
+  end in
+  fn 0 n f
+
+let length t =
+  let rec fn t i = begin
+    match t with
+    | (lazy Nil) -> i
```

Comment:
Lots of cases of unneeded parenthesis around (lazy foo). These should be removed.